src/App.tsx

```diff
@@ -1,13 +1,12 @@
 import React from "react";
```

Comment:
you dont need an app file

only index that will export the components so you can import for others who use your library

index.ts:
export { Button } from "./components/button/Button";

an so on...

in root open folder '.storybook':
config.js:
import { addDecorator } from "@storybook/react";
import themeDecorator from "./themeDecorator";

addDecorator(themeDecorator);


main.js:
const path = require("path");

module.exports = {
  stories: ["../src/**/*.stories.tsx"],
  // Add any Storybook addons you want here: https://storybook.js.org/addons/
  addons: [],
  webpackFinal: async (config) => {
    config.module.rules.push({
      test: /\.scss$/,
      use: ["style-loader", "css-loader", "sass-loader"],
      include: path.resolve(__dirname, "../")
    });

    config.module.rules.push({
      test: /\.(ts|tsx)$/,
      loader: require.resolve("babel-loader"),
      options: {
        presets: [["react-app", { flow: false, typescript: true }]]
      }
    });
    config.resolve.extensions.push(".ts", ".tsx");

    return config;
  }
};

themeDecorator.js:
import React from "react"
import { ThemeProvider } from '../src';

const ThemeDecorator = storyFn => (
  <ThemeProvider>{storyFn()}</ThemeProvider>
)

export default ThemeDecorator

Reply:
done.